public/app/features/integrated-alerting/components/AlertRules/AlertRulesTable/AlertRulesTable.tsx

```diff
@@ -1,78 +1,141 @@
 import React, { FC, useContext } from 'react';
-import { useTable } from 'react-table';
+import { TableState, useTable, usePagination } from 'react-table';
 import { Spinner, useStyles } from '@grafana/ui';
 import { css } from 'emotion';
 import { getStyles } from './AlertRulesTable.styles';
 import { AlertRule } from '../AlertRules.types';
 import { AlertRulesTableProps } from './AlertRulesTable.types';
 import { AlertRulesProvider } from '../AlertRules.provider';
+import { Pagination } from '../../Table/Pagination';
+import { ExtendedTableInstance, ExtendedTableOptions } from '../../Table';
+import { PAGE_SIZES } from 'app/core/constants';
 
-export const AlertRulesTable: FC<AlertRulesTableProps> = ({ pendingRequest, data, columns, emptyMessage }) => {
+export const AlertRulesTable: FC<AlertRulesTableProps> = ({
+  pendingRequest,
+  data,
+  columns,
+  emptyMessage,
+  showPagination,
+  totalPages,
+  totalItems,
+  pageSize: propPageSize,
+  pageIndex: propPageIndex,
+  onPaginationChanged,
+}) => {
   const style = useStyles(getStyles);
   const { selectedRuleDetails } = useContext(AlertRulesProvider);
-  const tableInstance = useTable({ columns, data });
-  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = tableInstance;
+  const manualPagination = totalPages >= 0;
+  const initialState: Partial<TableState> = {
+    pageIndex: propPageIndex,
+    pageSize: propPageSize,
+  } as Partial<TableState>;
```

Comment:
No need to have `Partial<TableState>` in two places, one can be removed.

Reply:
I'd remove the cast (`as`)

Reply:
actually TS only accepts `as` in this case, had to let go the other one